ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/QueryBackend.scala

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.dbbackend
+
+import scalaz.syntax.std.option._
+import scalaz.syntax.std.string._
+
+import QueryBackend.RawConf
+import OracleQueries.DisableContractPayloadIndexing
+
+private[http] sealed abstract class QueryBackend {
+  type SqlInterpol
+  // Why does the difference between QueryBackend and Queries matter at all?
+  // Because we cannot talk about the `Conf` of a (String => Queries) without
+  // introducing another existential scope, which is not only a bit mind-twisting,
+  // but won't work at all in Scala 3.  So we introduce the scope *here*, with
+  // the added benefit that we can leave only the purely existential interactions
+  // with `Queries` as that type's responsibility.
+  type Conf
+
+  def queries(tablePrefix: String, conf: Conf)(implicit sqli: SqlInterpol): Queries
+
+  def parseConf(kvs: RawConf): Either[String, Conf]
+}
+
+private[http] object QueryBackend {
+  type RawConf = Map[String, String]
+  type Aux[SqlI] = QueryBackend { type SqlInterpol = SqlI }
+  type Aux2[SqlI, Conf0] = QueryBackend { type SqlInterpol = SqlI; type Conf = Conf0 }
```

Comment:
What is the use of the Aux types here?

Reply:
I think I understood that part too, in the end it's just a type alias to be able to write type signatures shorter.

Reply:
There's another reason, too. It's common to have a type `Conf` in scope, and now I want to say "the `QueryBackend` with the same `Conf` type. But `{ type Conf = Conf }` doesn't refer to the `Conf` I already have in scope, it's a circular reference. With `Aux`, [I can do this just fine](https://github.com/digital-asset/daml/blob/e5c47347809562bdd2b5e24c75105c0adaf38794/ledger-service/http-json/src/main/scala/com/digitalasset/http/dbbackend/SupportedJdbcDriver.scala#L38-L40).